sdk/test-utils/perf/src/workerProgram.ts

```diff
@@ -75,15 +76,16 @@ export class WorkerPerfProgram implements PerfProgram {
     await Promise.all(this.tests.map((test) => test.postSetup?.()));
     await exitStage("postSetup");
 
-    if (this.options.warmup.value! > 0) {
+    if (this.options.warmup.value > 0) {
       await enterStage("warmup");
       await this.runTests(this.options.warmup.value!);
       await exitStage("warmup");
     }
 
-    for (let iteration = 0; iteration < this.options.iterations.value!; ++iteration) {
+    for (let iteration = 0; iteration < this.options.iterations.value; ++iteration) {
       await enterStage("test");
-      await this.runTests(this.options.duration.value!);
+      const duration = this.options.duration.value;
+      await monitorFunc(() => this.runTests(duration), this.options.profile.value, this.options["profile-filepath"].value);
```

Comment:
I'd only wrap the function in `monitorFunc` if profiling was requested.

Reply:
That's what it does internally, I should probably rename or improve readability.